### PR TITLE
Adding Secure Tunnel Sample to CI tests in Codebuild

### PR DESCRIPTION
**Pull request by sbSteveK**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: codebuild/samples/securetunnel-linux.sh [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+echo "aws iotsecuretunneling open-tunnel"
+RESPONSE=$(aws iotsecuretunneling open-tunnel)
+array=(${RESPONSE//:/ })
+
+echo "Assigning SOURCETOKEN"
+SOURCETOKEN="${array[11]}"
+
+echo "SOURCETOKEN"
+echo $SOURCETOKEN
```
Comment (bretambrose) [inline codebuild/samples/securetunnel-linux.sh:22]: Can we remove the two token echoes?

File: samples/secure_tunneling/secure_tunnel/main.cpp [hunk 2]
```diff
@@ -78,6 +78,16 @@ int main(int argc, char *argv[])
     string proxyUserName;
     string proxyPassword;
 
+    /*
+     * For internal testing
+     */
+    bool isTest = false;
+    int testCount = 5;
```
Comment (bretambrose) [inline samples/secure_tunneling/secure_tunnel/main.cpp:85]: Feel like this name could be a little clearer (expectedMessageCount? or similar)

File: samples/secure_tunneling/secure_tunnel/main.cpp [hunk 3]
```diff
@@ -78,6 +78,16 @@ int main(int argc, char *argv[])
...
+    if (s_cmdOptionExists(argv, argv + argc, "--test"))
```
Comment (bretambrose) [inline samples/secure_tunneling/secure_tunnel/main.cpp:86]: bool isTest = s_cmdOptionExists(argv, argv + argc, "--test");

maybe

Comment (sbSteveK) [reply samples/secure_tunneling/secure_tunnel/main.cpp]: This is so much cleaner. Thanks!

**Review by bretambrose — APPROVED** [review]

**Pull request merged**